runbot_docker_orchest/tests/test_runbot_build.py

```diff
@@ -0,0 +1,104 @@
+from odoo.tests.common import TransactionCase
+from odoo.tools.misc import mute_logger
+import xmlrpc.client
+import logging
+import time
+import unittest
+import os
+
+_logger = logging.getLogger(__name__)
+
+DOCKER_LOGIN = (
+    os.environ.get('DOCKER_PASSWORD') and os.environ.get('DOCKER_USERNAME'))
+
+
+class TestRunbotJobs(TransactionCase):
+
+    def setUp(self):
+        super(TestRunbotJobs, self).setUp()
+        self.build_obj = self.env['runbot.build']
+        self.repo_obj = self.env['runbot.repo']
+        self.branch_obj = self.env['runbot.branch']
+        self.repo = self.repo_obj.search([
+            ('is_docker_image', '=', True)], limit=1)
+        self.repo_domain = [('repo_id', '=', self.repo.id)]
+        self.cron = self.env.ref('runbot.runbot_repo_cron')
+        self.cron.write({'active': False})
+        self.build_obj.search([]).unlink()
+        self.build = None
+
+    @unittest.skipIf(not DOCKER_LOGIN,
+                     "If real values are not used then skip")
+    @mute_logger('odoo.addons.runbot.models.repo')
+    def test_jobs(self):
+        'Create build and run all jobs'
+        self.assertEqual(len(self.repo), 1, "Repo not found")
+        _logger.info("Repo update to get branches")
+        self.repo._update(self.repo)
+        branch = self.branch_obj.search(self.repo_domain + [
+            ('name', '=', 'latest')], limit=1)
+        self.assertTrue(branch, "Branch not found")
+        self.build = self.build_obj.search([('branch_id', '=', branch.id)])
+        _logger.info("Repo Cron to change state to pending -> testing")
+        self.assertEqual(
+            self.build.state, u'pending', "State should be pending")
+        self.repo._cron()
```

Comment:
Download the image early like as runbot_travis2docker:
https://github.com/Vauxoo/runbot-addons/blob/92d79ae48e1ad3f30039a8ea99bb12a55f2751c0/runbot_travis2docker/tests/test_runbot_build.py#L116-L118

in order to about a job kill because spent too much time.